pages/docs/integrations/quarkus-langchain4j.mdx

```diff
@@ -0,0 +1,99 @@
+description: Learn how to integrate Langfuse with Quarkus LangChain4j using OpenTelemetry for enhanced observability and performance monitoring in your AI applications.
+---
+
+# Integrating Langfuse with Quarkus LangChain4j
+
+This guide shows how to integrate [Langfuse](/) with [Quarkus LangChain4j](https://docs.quarkiverse.io/quarkus-langchain4j/dev/index.html) using OpenTelemetry.
+
+**Quarkus LangChain4j**: A [Quarkus](https://quarkus.io/) based integration of [LangChain4j](https://docs.langchain4j.dev/) for AI development with built-in OTel tracing for AI calls.
+
+**Langfuse**: Open source LLM engineering platform for observability, evals and prompt management.
+
+<Callout type="info" emoji="🐞">
+Please raise an Issue on [GitHub](/issues) if you face any issues with this integration.
+</Callout>
+
+## Step 1: Enable OpenTelemetry in Quarkus LangChain4j
+
+**Add Quarkus OpenTelemetry Dependency** (Maven):
+
+For Maven, add the following to your `pom.xml` (Gradle users can include equivalent coordinates in Gradle):
+
+```xml
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-opentelemetry</artifactId>
+</dependency>
+```
+
+**Configure OpenTelemetry exporter** (`application.properties`):
+
+```properties
+quarkus.otel.exporter.otlp.traces.protocol=http/protobuf
+```
+
+With these configurations and dependencies in place, your Quarkus application is ready to produce OpenTelemetry traces. Quarkus LangChain4j internal calls (e.g. when you invoke a chat model) will be recorded as spans.
+
+Each span will carry attributes like `gen_ai.operation.name`, `gen_ai.system` (the provider, e.g. “openai”), model names, token usage, etc., and – since we enabled them – events for the prompt and response content
+
+## Step 2: Configure Langfuse
+
+Now that your Quarkus application is emitting OpenTelemetry trace data, the next step is to direct that data to Langfuse.
+
+Langfuse will act as the “backend” for OpenTelemetry in this setup – essentially replacing a typical Jaeger/Zipkin/OTel-Collector with Langfuse’s trace ingestion API.
+
+
+**Langfuse Setup**
+   - Sign up for [Langfuse Cloud](https://cloud.langfuse.com/) or [self-hosted Langfuse](https://langfuse.com/self-hosting).
+   - Set the OTLP endpoint (e.g. `https://cloud.langfuse.com/api/public/otel`) and API keys.
+
+Configure these via environment variables:
+
+```bash
+QUARKUS_OTEL_EXPORTER_OTLP_ENDPOINT: set this to the Langfuse OTLP URL (e.g. https://cloud.langfuse.com/api/public/otel).
+QUARKUS_OTEL_EXPORTER_OTLP_HEADERS: set this to Authorization=Basic <base64 public:secret>.
+```
+
+<Callout type="info">
+You can find more on authentication via Basic Auth [here](/docs/opentelemetry/get-started).
+</Callout>
+
+## Step 3: Run a Test AI Operation
+
+Start your Quarkus application. Trigger an AI operation that Quarkus LangChain4j handles – for example, call a service or controller that uses a `ChatModel` to generate a completion.
```

Comment:
Could you add a reference to https://github.com/langfuse/langfuse-examples/tree/main/applications/quarkus-langchain4j-demo here? I think that would help understanding the E2E case.